src/Mod/CAM/Path/Dressup/Array.py

```diff
@@ -166,8 +166,37 @@ def onChanged(self, obj, prop):
         if prop == "Type":
             self.setEditorModes(obj)
 
-    def dresupOnDocumentRestored(self, obj):
+    def onDocumentRestored(self, obj):
         """onDocumentRestored(obj) ... Called automatically when document is restored."""
+
+        # Before commit 24c32c9 array dressups did have their own CoolantMode
+        # and ToolController which they copied from the base operation. We need
+        # to remove those properties when loading a file that was created before
+        # that commit. Otherwise the wrong or no CoolantMode/ToolController will
+        # be used.
+
+        if hasattr(obj, "CoolantMode"):
+            obj.removeProperty("CoolantMode")
+            FreeCAD.Console.PrintWarning(
+                translate(
+                    "DressupArray",
+                    "Removing CoolantMode property from {} as base operation's CoolantMode is now used.\n".format(
+                        obj.Name
+                    ),
+                )
+            )
```

Comment:
Strings need to be static for `translate()` to work.
It's also good to move out new line at the end, as translators might remove it by accident.

```suggestion
            translation = translate(
                "DressupArray",
                "Removing CoolantMode property from {} as base operation's CoolantMode is now used.",
            ) + "\n"
            FreeCAD.Console.PrintWarning(translation.format(obj.Name))
```

Reply:
Oh yes makes sense. Fixed. 👍